crates/typst/src/text/smartquote.rs

```diff
@@ -168,10 +168,14 @@ impl Default for SmartQuoter {
     }
 }
 
-fn is_ignorable(c: char) -> bool {
+fn is_exterior_to_quote(c: char) -> bool {
     c.is_whitespace() || is_newline(c)
 }
 
+pub(crate) fn is_ignorable(c: char) -> bool {
+    icu_properties::sets::default_ignorable_code_point().contains(c)
```

Comment:
currently, we don't use the compiled_data from icu4x, but rather generated one (see [here for how it works](https://github.com/typst/typst/blob/59374f737079472aa5694b69ebced8c4cd5f9dfc/crates/typst/src/layout/inline/linebreak.rs#L55-L59)) and [here for the reason](https://github.com/typst/typst/pull/4485#issuecomment-2211732974). This might change in the future. Either we investigate whether we should change that before merging this PR or it should switch to the blob data.

Reply:
Changed to use the generated `DEFAULT_IGNORABLE_DATA`.